solidity/dashboard/src/components/AddressShortcut.jsx

```diff
@@ -10,14 +11,22 @@ const AddressShortcut = ({ address, classNames }) => {
       toCopy={address}
       render={({ reset, copyStatus, copyToClipboard }) => {
         return (
-          <button
-            onClick={copyToClipboard}
-            onMouseOut={reset}
-            className={`address-shortcut tooltip address ${classNames}`}
+          <Tooltip
+            simple
+            triggerComponent={() => {
```

Comment:
Hmm to be honest not follow. A Tooltip component handle all "logic" about showing and hiding tooltip. So we just pass a trigger component instead of handling feature outside tooltip component.

In that case you described we would have to pass a component to render to `showCopy` function and take care of hiding/showing every time when we want to use tooltip component.

Reply:
My conceptualization here is that *where the tooltip is triggered* is not a concern of the tooltip, it's a concern of the caller using the tooltip—something I feel is hinted at by the hierarchy, where the caller is above the tooltip, and the trigger is never below the tooltip. In this formulation, the tooltip component provides all the tools to present tooltips, including functions to show and hide it, but actually hooking that up is something left to the caller.